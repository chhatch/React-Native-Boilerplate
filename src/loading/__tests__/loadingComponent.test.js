import * as React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'

import LoadingComponent from '../component'

const mockStore = configureStore([])

describe('Loading component', () => {
    it('should show loading if loading', () => {
        store = mockStore({
            loading: {
                loading: true,
            },
        })
        const tree = renderer
            .create(
                <Provider store={store}>
                    <LoadingComponent />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should not show loading if not loading', () => {
        store = mockStore({
            loading: {
                loading: false,
            },
        })
        const tree = renderer
            .create(
                <Provider store={store}>
                    <LoadingComponent />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
