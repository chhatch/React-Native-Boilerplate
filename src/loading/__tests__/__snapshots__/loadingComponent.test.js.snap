// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should not show loading if not loading 1`] = `null`;

exports[`Loading component should show loading if loading 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="black"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
